aggregator/src/aggregator/aggregation_job_continue.rs

```diff
@@ -59,201 +56,216 @@ impl VdafOps {
                                     prepare_error: PrepareError::ReportDropped,
                                 })
                                 .with_last_prep_resp(None),
                             None,
                         ));
                     }
+                    continue;
+                }
+                break report_agg;
+            };
+
+            let prep_state = match report_aggregation.state() {
+                ReportAggregationState::WaitingHelper { prepare_state } => prepare_state.clone(),
+                ReportAggregationState::WaitingLeader { .. } => {
+                    return Err(datastore::Error::User(
+                        Error::Internal(
+                            "helper encountered unexpected ReportAggregationState::WaitingLeader"
+                                .to_string(),
+                        )
+                        .into(),
+                    ))
+                }
+                _ => {
+                    return Err(datastore::Error::User(
+                        Error::InvalidMessage(
+                            Some(*task.id()),
+                            "leader sent prepare step for non-WAITING report aggregation",
+                        )
+                        .into(),
+                    ));
                 }
-                Ok(report_aggregations_to_write)
+            };
+
+            prep_steps_and_ras.push((prep_step.clone(), report_aggregation, prep_state));
+        }
+
+        for report_aggregation in report_aggregations_iter {
+            // This report was omitted by the leader because of a prior failure. Note that
+            // the report was dropped (if it's not already in an error state) and continue.
+            if matches!(
+                report_aggregation.state(),
+                ReportAggregationState::WaitingHelper { .. }
+            ) {
+                report_aggregations_to_write.push(WritableReportAggregation::new(
+                    report_aggregation
+                        .clone()
+                        .with_state(ReportAggregationState::Failed {
+                            prepare_error: PrepareError::ReportDropped,
+                        })
+                        .with_last_prep_resp(None),
+                    None,
+                ));
+            }
+        }
+
+        // Compute the next aggregation step.
+        let (sender, mut receiver) = mpsc::unbounded_channel();
+        let aggregation_job = Arc::new(aggregation_job);
+        let producer_task = tokio::task::spawn_blocking({
+            let parent_span = Span::current();
+            let metrics = metrics.clone();
+            let task = Arc::clone(&task);
+            let vdaf = Arc::clone(&vdaf);
+            let aggregation_job = Arc::clone(&aggregation_job);
+
+            move || {
+                let span = info_span!(parent: parent_span, "step_aggregation_job threadpool task");
+                let _entered = span.enter();
```

Comment:
As with aggregation init, this span should be entered in the closure passed to `try_for_each_with()`.